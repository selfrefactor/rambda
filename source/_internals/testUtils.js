import combinate from 'combinate'
import {
  equals,
  filter,
  forEach,
  init,
  last,
  omit,
  switcher,
  type,
} from 'rambdax'

const omitOk = omit('ok')

const PENDING = 'PENDING'
const RESULTS_EQUAL = 'results are equal'
const RESULTS_MISMATCH = 'results are different'
const ERRORS_EQUAL = 'errors are equal'
const ERRORS_MESSAGE_MISMATCH = 'errors messages are different'
const ERRORS_TYPE_MISMATCH = 'errors types are different'
const SHOULD_THROW = 'Rambda should throw'
const SHOULD_NOT_THROW = 'Rambda should not throw'
const ALL_ERROR_LABELS = {
  RESULTS_MISMATCH,
  ERRORS_MESSAGE_MISMATCH,
  ERRORS_TYPE_MISMATCH,
  SHOULD_NOT_THROW,
  SHOULD_THROW,
}
const MISSING = 'Missing error handle in parseError'

function parseError(err){
  const typeError = switcher(err)
    .is(x => x instanceof TypeError, 'TypeError')
    .is(x => x instanceof SyntaxError, 'SyntaxError')
    .is(x => x instanceof RangeError, 'RangeError')
    .is(x => x instanceof Error, 'Error')
    .default(MISSING)

  if (typeError === MISSING){
    throw new Error('typeError === MISSING')
  }

  return {
    message : err.message,
    type    : typeError,
    ok      : true,
  }
}

function executeSync(
  fn, inputs, returnsFunctionFlag
){
  let result = PENDING
  let error = { ok : false }
  try {
    result = returnsFunctionFlag ?
      fn(...init(inputs))(last(inputs)) :
      fn(...inputs)
  } catch (e){
    error = parseError(e)
  }

  return {
    result,
    error,
  }
}

async function executeAsync(
  fn, inputs, returnsFunctionFlag
){
  let result = PENDING
  let error = { ok : false }
  try {
    result = returnsFunctionFlag ?
      await fn(...init(inputs))(last(inputs)) :
      await fn(...inputs)
  } catch (e){
    error = parseError(e)
  }

  return {
    result,
    error,
  }
}

export function profileMethod({
  firstInput,
  secondInput = undefined,
  thirdInput = undefined,
  returnsFunctionFlag = false,
  fn,
}){
  const combinationsInput = filter(Boolean, {
    firstInput,
    secondInput,
    thirdInput,
  })
  const inputKeys = Object.keys(combinationsInput)
  const combinations = combinate(combinationsInput)

  combinations.forEach(combination => {
    const inputs = [
      combination.firstInput,
      combination.secondInput,
      combination.thirdInput,
    ].filter((_, i) => i < inputKeys.length)

    test(getTestTitle(...inputs), () => {
      const { result, error } = executeSync(
        fn, inputs, returnsFunctionFlag
      )

      expect({
        result,
        error : error.ok ? omitOk(error) : PENDING,
        inputs,
      }).toMatchSnapshot()
    })
  })
}

export function profileMethodAsync({
  firstInput,
  secondInput = undefined,
  thirdInput = undefined,
  returnsFunctionFlag = false,
  fn,
}){
  const combinationsInput = filter(Boolean, {
    firstInput,
    secondInput,
    thirdInput,
  })
  const inputKeys = Object.keys(combinationsInput)
  const combinations = combinate(combinationsInput)

  combinations.forEach(combination => {
    const inputs = [
      combination.firstInput,
      combination.secondInput,
      combination.thirdInput,
    ].filter((_, i) => i < inputKeys.length)
    test(getTestTitle(...inputs), async () => {
      const { result, error } = await executeAsync(
        fn,
        inputs,
        returnsFunctionFlag
      )

      expect({
        result,
        error : error.ok ? omitOk(error) : PENDING,
        inputs,
      }).toMatchSnapshot()
    })
  })
}

export function compareToRamda(
  fn, fnRamda, returnsFunctionFlag
){
  return (...inputs) => {
    const { result, error } = executeSync(
      fn, inputs, returnsFunctionFlag
    )
    const { result: ramdaResult, error: ramdaError } = executeSync(
      fnRamda,
      inputs,
      returnsFunctionFlag
    )

    const toReturn = {
      result,
      ramdaResult,
      ramdaError : ramdaError.ok ? omitOk(ramdaError) : PENDING,
      error      : error.ok ? omitOk(error) : PENDING,
    }

    if (result !== PENDING){
      if (ramdaError.ok){
        return {
          ...toReturn,
          ok    : false,
          label : SHOULD_THROW,
        }
      }

      if (equals(result, ramdaResult)){
        return {
          ...toReturn,
          ok    : true,
          label : RESULTS_EQUAL,
        }
      }

      return {
        ...toReturn,
        ok    : false,
        label : RESULTS_MISMATCH,
      }
    }

    if (equals(error, ramdaError)){
      return {
        ...toReturn,
        ok    : true,
        label : ERRORS_EQUAL,
      }
    }

    if (ramdaError.ok){
      return {
        ...toReturn,
        ok    : false,
        label :
          ramdaError.type === error.type ?
            ERRORS_MESSAGE_MISMATCH :
            ERRORS_TYPE_MISMATCH,
      }
    }

    return {
      ...toReturn,
      ok    : false,
      label : SHOULD_NOT_THROW,
    }
  }
}

export const getTestTitle = (...inputs) => inputs.map(type).join(' | ')

export const compareCombinations = ({
  firstInput,
  secondInput = undefined,
  thirdInput = undefined,
  returnsFunctionFlag = false,
  setCounter = () => {},
  callback = x => {},
  fn,
  fnRamda,
}) => {
  const counter = {
    RESULTS_MISMATCH        : 0,
    SHOULD_THROW            : 0,
    SHOULD_NOT_THROW        : 0,
    ERRORS_TYPE_MISMATCH    : 0,
    ERRORS_MESSAGE_MISMATCH : 0,
    TOTAL_TESTS             : 0,
  }

  const increaseCounter = comparedResult => {
    if (comparedResult.ok) return
    let counterProp
    forEach((x, prop) => {
      if (x === comparedResult.label) counterProp = prop
    }, ALL_ERROR_LABELS)
    counter[ counterProp ]++
  }

  const combinationsInput = filter(Boolean, {
    firstInput,
    secondInput,
    thirdInput,
  })
  const inputKeys = Object.keys(combinationsInput)
  const combinations = combinate(combinationsInput)
  const compareOutputs = compareToRamda(
    fn, fnRamda, returnsFunctionFlag
  )

  afterAll(() => callback(counter))

  combinations.forEach(combination => {
    const inputs = [
      combination.firstInput,
      combination.secondInput,
      combination.thirdInput,
    ].filter((_, i) => i < inputKeys.length)

    test(getTestTitle(...inputs), () => {
      const compared = compareOutputs(...inputs)
      setCounter()
      if (!compared.ok){
        // if (compared.label === RESULTS_MISMATCH){
        increaseCounter(compared)
        expect({
          ...compared,
          inputs,
        }).toMatchSnapshot()
      }
      counter.TOTAL_TESTS++
    })
  })
}

const demoMap = new Map()

demoMap.set('a', 1)
demoMap.set('b', 2)

const demoWeakMap = new WeakMap()

demoWeakMap.set({ a : 0 }, 1)
demoWeakMap.set({ b : 0 }, 2)

const demoWeakSet = new WeakSet()
const foo = { a : 1 }
const bar = { b : 2 }

demoWeakSet.add(foo)
demoWeakSet.add(bar)

const demoBigInt = BigInt(9007199254740991)

const demoRegExp = new RegExp('ab+c', 'i')
const demoSymbol = Symbol('foo')

export const FALSY_VALUES = [ null, undefined, false, NaN, '' ]
export const EXTRA_BUILD_IN_OBJECTS = [
  Infinity,
  -Infinity,
  Symbol,
  new ArrayBuffer(8),
  demoRegExp,
  demoMap,
  demoWeakMap,
  demoWeakSet,
  demoBigInt,
  demoSymbol,
]
