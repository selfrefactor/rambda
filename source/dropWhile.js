export function dropWhile(predicate) {
  return iterable => {
    const toReturn = []
    let counter = 0

    while (counter < iterable.length) {
      const item = iterable[counter++]
      if (!predicate(item, counter)) {
        toReturn.push(item)
        break
      }
    }

    while (counter < iterable.length) {
      toReturn.push(iterable[counter++])
    }

    return toReturn
  }
}
