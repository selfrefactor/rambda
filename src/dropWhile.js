import {_isArray} from '../src/_internals/_isArray'

export function dropWhile(predicate, iterable) {
  if (arguments.length === 1) {
    return _iterable => dropWhile(predicate, _iterable)
  }
  const isArray = _isArray(iterable)
  if (!isArray && typeof iterable !== 'string') {
    throw new Error('`iterable` is neither list nor a string')
  }
  let flag = false
  const holder = []
  let counter = -1

  while (counter++ < iterable.length - 1) {
    if (flag) {
      holder.push(iterable[counter])
    } else if (!predicate(iterable[counter])) {
      if (!flag) flag = true

      holder.push(iterable[counter])
    }
  }

  return isArray ? holder : holder.join('')
}
