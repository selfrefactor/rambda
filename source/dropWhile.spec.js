import { dropWhile as dropWhileRamda } from 'ramda'

import { compareCombinations } from './_internals/testUtils.js'
import { dropWhile } from './dropWhile.js'

const list = [ 1, 2, 3, 4 ]

test('happy', () => {
  const predicate = x => x < 3
  const result = dropWhile(predicate, list)
  expect(result).toEqual([ 3, 4 ])
})

test('always true', () => {
  const predicate = () => true
  const result = dropWhileRamda(predicate, list)
  expect(result).toEqual([])
})

test('always false', () => {
  const predicate = () => 0
  const result = dropWhile(predicate)(list)
  expect(result).toEqual(list)
})

test('works with string as iterable', () => {
  const iterable = 'foobar'
  const predicate = x => x !== 'b'
  const result = dropWhile(predicate, iterable)
  expect(result).toBe('bar')
})

const possiblePredicates = [
  null,
  undefined,
  () => 0,
  () => true,
  /foo/g,
  {},
  [],
]

const possibleIterables = [
  null,
  undefined,
  [],
  {},
  1,
  '',
  'foobar',
  [ '' ],
  [ 1, 2, 3, 4, 5 ],
]

describe('brute force', () => {
  compareCombinations({
    firstInput : possiblePredicates,
    callback   : errorsCounters => {
      expect(errorsCounters).toMatchInlineSnapshot(`
        {
          "ERRORS_MESSAGE_MISMATCH": 15,
          "ERRORS_TYPE_MISMATCH": 14,
          "RESULTS_MISMATCH": 0,
          "SHOULD_NOT_THROW": 14,
          "SHOULD_THROW": 0,
          "TOTAL_TESTS": 63,
        }
      `)
    },
    secondInput : possibleIterables,
    fn          : dropWhile,
    fnRamda     : dropWhileRamda,
  })
})
