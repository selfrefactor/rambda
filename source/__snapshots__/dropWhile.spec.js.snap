// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`brute force Array | Array 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    [],
    [
      "",
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Array | Array 2`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    [],
    [
      1,
      2,
      3,
      4,
      5,
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Array | Null 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [],
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Array | Number 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [],
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Array | Object 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [],
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Array | String 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    [],
    "foobar",
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Array | Undefined 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [],
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Function | Null 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Function | Null 2`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Function | Number 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Function | Number 2`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Function | Object 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Function | Object 2`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Function | Undefined 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Function | Undefined 2`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    [Function],
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Null | Array 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    null,
    [
      "",
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Null | Array 2`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    null,
    [
      1,
      2,
      3,
      4,
      5,
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Null | Null 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    null,
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Null | Number 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    null,
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Null | Object 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    null,
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Null | String 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    null,
    "foobar",
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Null | Undefined 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    null,
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Object | Array 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    {},
    [
      "",
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Object | Array 2`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    {},
    [
      1,
      2,
      3,
      4,
      5,
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Object | Null 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    {},
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Object | Number 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    {},
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Object | Object 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    {},
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Object | String 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    {},
    "foobar",
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Object | Undefined 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    {},
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force RegExp | Array 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    /foo/g,
    [
      "",
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force RegExp | Array 2`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    /foo/g,
    [
      1,
      2,
      3,
      4,
      5,
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force RegExp | Null 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    /foo/g,
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force RegExp | Number 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    /foo/g,
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force RegExp | Object 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    /foo/g,
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force RegExp | String 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    /foo/g,
    "foobar",
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force RegExp | Undefined 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    /foo/g,
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Undefined | Array 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    undefined,
    [
      "",
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Undefined | Array 2`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    undefined,
    [
      1,
      2,
      3,
      4,
      5,
    ],
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Undefined | Null 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    undefined,
    null,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of null (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Undefined | Number 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    undefined,
    1,
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Undefined | Object 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    undefined,
    {},
  ],
  "label": "Rambda should not throw",
  "ok": false,
  "ramdaError": "PENDING",
  "ramdaResult": [],
  "result": "PENDING",
}
`;

exports[`brute force Undefined | String 1`] = `
{
  "error": {
    "message": "predicate is not a function",
    "type": "TypeError",
  },
  "inputs": [
    undefined,
    "foobar",
  ],
  "label": "errors messages are different",
  "ok": false,
  "ramdaError": {
    "message": "pred is not a function",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;

exports[`brute force Undefined | Undefined 1`] = `
{
  "error": {
    "message": "\`iterable\` is neither list nor a string",
    "type": "Error",
  },
  "inputs": [
    undefined,
    undefined,
  ],
  "label": "errors types are different",
  "ok": false,
  "ramdaError": {
    "message": "Cannot read properties of undefined (reading 'dropWhile')",
    "type": "TypeError",
  },
  "ramdaResult": "PENDING",
  "result": "PENDING",
}
`;
